app/services/more_like_this_getter.rb

```diff
@@ -0,0 +1,121 @@
+# MoreLikeThisGetter.new(work).works
+#
+# returns an array of zero to ten
+# published works that SOLR thinks
+# are "like" the given work.
+# The array is in order: the first item in
+# the one SOLR deemed most "alike".
+#
+# This calls the SOLR server's more-like-this request handler
+# at http://SOME_URL/solr/SOME_CORE/mlt
+#
+# See also:
+# work_indexer.rb for indexing details,
+# solr/config/solrconfig.xml for the request handler.
+#
+# We really do not want to slow down (or worse, break) the work show page
+# if SOLR is slow or unreachable, as it often is.
+# If anything goes wrong, you'll just get an empty array.
+#
+# MoreLikeThisGetter.new(work).works
+#
+class MoreLikeThisGetter
+  attr_reader :work
+
+  # These timeouts are short. We want to be conservative,
+  # as we don't want to trust our SOLR provider too much
+  # and this appears on a part of the website that's usually fast.
+  TIMEOUT=1
+  OPEN_TIMEOUT=1
+
+  # @param work [Work] Work
+  # @param max_number_of_works: if specified,
+  # this limits the number of works returned.
+  def initialize(work, max_number_of_works: nil)
+    @work = work
+    @max_number_of_works = max_number_of_works
+  end
+
+  # Returns an array of up to @max_number_of_works
+  # published works that SOLR deems similar, in order of similarity
+  def works
+    friendlier_ids.map {|id| works_in_arbitrary_order[id] }.compact
+  end
+
+  def works_in_arbitrary_order
+    @works_in_arbitrary_order ||= Work.where(
+      friendlier_id: friendlier_ids,
+      published: true)&.
+      index_by(&:friendlier_id) || {}
+  end
+
+  # Some justification for the choices SOLR made.
+  # Does not result in an additional call to SOLR.
+  def json
+    more_like_this_doc_set&.map do |doc|
+      doc.select do |key, value|
+        key == 'text1_tesim' || key.include?('more_like_this')
+      end
+    end
+  end
+
+  # Returns an RSolr::Client configured with our
+  # standard blacklight config, except it won't retry failed queries.
+  def solr_connection
+    # If we don't get a response from SOLR right away,
+    # we just want to show the page without the more_like_this content. 
+    #
+    # Note the existence of Scihist::BlacklightSolrRepository, which we are
+    # consciously not using as the solr repo in this method. Its only purpose
+    # is to provide two successive retries on failure, which we don't really want here.
+    @solr_connection ||= begin
+      repo_class = Blacklight::Solr::Repository
+      faraday_params = {
+        :timeout => TIMEOUT,
+        :open_timeout => OPEN_TIMEOUT
+      }
+      repo_class.new(CatalogController.blacklight_config).connection.tap do |conn|
+        conn.connection.params = faraday_params
+      end
```

Comment:
Phew, tricky abstract blacklight stuff, nice job figuring it out. 

I think I'd use `Scihist::BlacklightSolrRepository` to get the retries, why don't we want the retries? 

Either way, to me this would be more clear without the local variables, just say what you mean all at once:

```ruby
Blacklight::Solr::Repository.new(CatalogController.blacklight_config).connection.tap do |conn|
   conn.connection_params = {
         :timeout => TIMEOUT,
        :open_timeout => OPEN_TIMEOUT
   }
end
```

But now I also wonder if we could skip the custom timeouts... why not just use the default solr connection with all it's retries and default timeouts, to make it as simple as:

```ruby
Scihist::BlacklightSolrRepository.new(CatalogController.blacklight_config).connection
```

Or if you really want the custom timeout and no-retry-behavior, ok, cool, as above.

Reply:
Yeah, I feel like `more_like_this`  is nice to have, but if SOLR is slow or unavailable I want to prevent any slowdown in the display of the work show page.

Reply:
ok fair!